riscv-sbi.adoc

```diff
@@ -1674,6 +1676,334 @@ The possible error codes returned in `sbiret.error` are shown in the
+* A https://oidref.com/2.5.29.49[UserNotice] X.509 extension containing the
+the data blob from `request_data_addr`.
+* A https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.1[AuthorityKeyIdentifier]
+X.509 extension.
+
+The possible error codes returned in `sbiret.error` are shown in the
+<<table_get_evidence_errors>> below.
+
+[#table_get_evidence_errors]
+.Attestation Get Evidence Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code          | Description
+| SBI_SUCCESS         | The evidence is generated and the certificate size is
+                        `sbiret.value`.
+| SBI_ERR_FAILED      | The evidence could not be generated.
+| SBI_INVALID_ADDRESS | The CSR or the certificate adress is invalid.
+| SBI_INVALID_PARAM   | The CSR at `csr_addr` is invalid.
+| SBI_ERR_BUSY        | The evidence could not be generated due to some
+                        resources being busy. The request may be retried.
+|===
+
+[NOTE]
+.Attestation Flow Example
+====
+Guest userspace flow TBD
+====
+
+[#measurement-extend-function,reftext="Extend Measurement"]
+=== Function: Extend Measurement (FID #1)
+
+[source, C]
+----
+struct sbiret sbi_measurement_extend(unsigned long measurement_addr,
+                                     unsigned long measurement_index);
+----
+
+Extend the platform TCB measurement with one additional measurement.
+
+This function allows the caller to add a software component to the list of the
+TCB layers measurements at any point of the platform lifecycle.
+This function provides a
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[DRTM]-like
+capability by allowing the caller to add a software component to the list of the
+TCB layers measurements at any point of the platform lifecycle.
+
+The SBI implementation should manage measurement added through this SBI function
+separately. It can choose to maintain them through multiple measurement
+registers, or extend a single one for every `sbi_attestation_extend_measurement`
+call. In the former case, the caller can select a measurement index through the
+`measurement_index` parameter to extend one particular measurement register.
+When supporting multiple measurement registers, SBI implementations should map
+the supported indexes to the
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[TCG specified DTRM PCR indexes].
+
+All measurements added through this SBI function must be included in the TCB
+attestation evidence generated by the <<get-evidence-function>> function.
+
+The measurement at the `measurement_addr` physical address must be generated
+by a hash function that implements the algorithm reported by the
+`HASH_ALGORITHM` field from the <<get-capabilities-function>> returned structure
+(`ATST_CAPS`). See <<table_attestation_and_measurement_capabilities_format>> for
+more details.
```

Comment:
As this function depends on get_capabilities function, that should be described first.

Reply:
I moved the `get_capabilities` function as the first FID, because the 2 other ones depend on it.